src/psyclone/psyGen.py

```diff
@@ -6070,8 +6142,41 @@ def __str__(self):
             ret = ret[:-2] + "]"  # Deletes last ", " and adds "]"
         else:
             ret += "Scalar"
+        if self.is_constant:
+            ret += ", constant, value={0}".format(self.constant_value)
         return ret + ">"
 
+    def copy(self):
+        '''Create and return a copy of this object.
+
+        :returns: A symbol object with the same properties as this \
+        symbol object
```

Comment:
Punctuation

Reply:
All together now, "Are you @TeranIvy in disguise ..."

Reply:
:)